GVFS/FastFetch/NativeUnixMethods.cs

```diff
@@ -14,21 +14,35 @@ public class NativeUnixMethods
         public const int Create = 0x0200;
         public const int Truncate = 0x0400;
         private const int InvalidFileDescriptor = -1;
+        private const ushort SymLinkMode = 0xA000;
 
         public static unsafe void WriteFile(ITracer tracer, byte* originalData, long originalSize, string destination, ushort mode)
         {
             int fileDescriptor = InvalidFileDescriptor;
             try
             {
-                fileDescriptor = Open(destination, WriteOnly | Create | Truncate, mode);
-                if (fileDescriptor != InvalidFileDescriptor)
+                if (mode == SymLinkMode)
                 {
-                    IntPtr result = Write(fileDescriptor, originalData, (IntPtr)originalSize);
-                    if (result.ToInt32() == -1)
+                    string linkLocation = Marshal.PtrToStringUTF8(new IntPtr(originalData));
+                    int result = CreateSymLink(linkLocation, destination);
+                    if (result == -1)
                     {
+                        tracer.RelatedInfo("Failed to create symlink({0},{1})", linkLocation, destination);

```

Comment:
Why are we writing a message when `CreateSymLink` fails but not when `Open` or `Write` fail (below)?

Assuming this exception is caught higher up the stack and logged as an error you could include a custom message in the exception itself:

```
throw new Win32Exception(
    Marshal.GetLastWin32Error(),
    $""Failed to create symlink({linkLocation},{destination})");
```

Reply:
This was mostly for my local testing, but now that this is working, I like your suggestion and will use it all over this WriteFile method and will rework how the exceptions are worded.